MinPlatformPkg/Test/Library/TestPointLib/MmTestPointCommunication.c

```diff
@@ -255,21 +252,14 @@ MmTestPointMmiHandlerGetDataByOffset (
   // MmiHandlerTestPointCopyData().
   //
   SpeculationBarrier ();
-  MmiHandlerTestPointCopyData (
-    Data,
-    DataSize,
-    (VOID *)(UINTN)MmiHandlerTestPointGetDataByOffset.DataBuffer,
-    &MmiHandlerTestPointGetDataByOffset.DataSize,
-    &MmiHandlerTestPointGetDataByOffset.DataOffset
-    );
 
   CopyMem (

```

Comment:
If I understand correctly, the DXE side caller is passing the size they got back in the `MMI_HANDLER_TEST_POINT_COMMAND_GET_INFO`. The necessary destination buffer size is gotten again here a little above by calling `GetAllMmTestPointData()`. It would be good to check that the size the user sent through the comm buffer is >= `DataSize` here before calling `CopyMem()`.

Reply:
I agree that would fix the possible time of check vs time of use problem.  It'll be fixed in the next commit.

Reply:
It's fixed now.